docs/walrus-sites/routing.md

```diff
@@ -0,0 +1,123 @@
+# Specifying headers and routing
+
+``` admonish tip title="New with Walrus Sites testnet version"
+The following features have been released with the Walrus Sites testnet version.
+```
+
+In its base configuration, Walrus Sites serves static assets through a Portal. However, many modern
+web applications require more advanced features, such as custom headers and client-side routing.
+
+Therefore, the site-builder can read a `ws-resource.json` configuration file, in which you can
+directly specify resource headers and routing rules.
+
+## The `ws-resources.json` file
+
+This file is optionally placed in the root of the site directory, and it is *not* uploaded with the
+site's resources (in other words, the file is not part of the resulting Walrus Site and is not
+served by the Portal).
+
+If you don't want to use this default location, you can specify the path to the configuration file
+with the `--ws-resources` flag when running the `publish` or `update` commands.
+
+The file is JSON-formatted, and looks like the following:
+
+``` JSON
+{
+  "headers": {
+    "/index.html": {
+      "Content-Type": "text/html; charset=utf-8",
+      "Cache-Control": "max-age=3500"
+    }
+  },
+  "routes": {
+    "/*": "/index.html",
+    "/accounts/*": "/accounts.html",
+    "/path/assets/*": "/assets/asset_router.html"
+  }
+}
+```
+
+We now describe in details the two sections of the configuration file, `headers` and `routes`.
+
+## Specifying HTTP response headers
+
+The `headers` section allows you to specify custom HTTP response headers for specific resources.
+The keys in the `headers` object are the paths of the resources, and the values are lists of
+key-value pairs corresponding to the headers that the Portal will attach to the response.
+
+For example, in the configuration above, the file `index.html` will be served with the
+`Content-Type` header set to `text/html; charset=utf-8` and the `Cache-Control` header set to
+`max-age=3500`.
+
+This mechanism allows you to control various aspects of the resource delivery, such as caching,
+encoding, and content types.
+
+```admonish
+The resource path is always represented as starting from the root `/`.
+```
+
+### Default headers
+
+By default, no headers need to be specified, and the `ws-resources.json` file can be omitted. The
+site-builder will automatically try to infer the `Content-Type` header based on the file extension,
+and set the `Content-Encoding` to `identity` (no transformation).
+
+In case the content type cannot be inferred, the `Content-Type` will be set to
+`application/octet-stream`.
```

Comment:
Still skeptical about describing implementation details here. 

This is not something that will change anytime soon so it's fine I guess.

Reply:
i think it's useful for devs that use weird extensions and then wonder why they have `octet-stream` headers coming in